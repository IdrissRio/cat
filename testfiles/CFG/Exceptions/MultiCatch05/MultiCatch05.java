class MultiCatch05 {
  void test() {
    try {
      try {
        m();
        int m;
      } catch (TestException1 e) {
        int n;
      }
    } catch (TestException1 | TestException3 e) {
      int a;
    } catch (TestException2 | TestException4 e) {
      int b;
    };
  }

  void m() throws TestException1, TestException2 {}
}

class TestException1 extends RuntimeException {}
class TestException2 extends RuntimeException {}
class TestException3 extends RuntimeException {}
class TestException4 extends RuntimeException {}