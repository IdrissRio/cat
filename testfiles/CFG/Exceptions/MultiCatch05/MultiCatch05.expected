NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[28] Successors: [MethodAccess[44]]
NodeName: MethodAccess[44] Successors: [VariableDeclarator[54], UncheckedExceptions[45]]
NodeName: VariableDeclarator[54] Successors: [UncheckedExceptions[74]]
NodeName: UncheckedExceptions[74] Successors: [UncheckedExceptions[111], Exit[35]]
NodeName: UncheckedExceptions[111] Successors: [Exit[35]]
NodeName: Exit[35] Successors: []
NodeName: UncheckedExceptions[45] Successors: [ParameterDeclaration[59], CatchParameterDeclaration[94], Exit[35]]
NodeName: ParameterDeclaration[59] Successors: [VariableDeclarator[70]]
NodeName: VariableDeclarator[70] Successors: [UncheckedExceptions[111]]
NodeName: CatchParameterDeclaration[94] Successors: [VariableDeclarator[107]]
NodeName: VariableDeclarator[107] Successors: [Exit[35]]
NodeName: Entry[114] Successors: [Exit[123]]
NodeName: Exit[123] Successors: []
NodeName: Entry[135] Successors: [Exit[136]]
NodeName: Exit[136] Successors: []
NodeName: Entry[127] Successors: [Exit[128]]
NodeName: Exit[128] Successors: []
NodeName: Entry[158] Successors: [Exit[159]]
NodeName: Exit[159] Successors: []
NodeName: Entry[150] Successors: [Exit[151]]
NodeName: Exit[151] Successors: []
NodeName: Entry[181] Successors: [Exit[182]]
NodeName: Exit[182] Successors: []
NodeName: Entry[173] Successors: [Exit[174]]
NodeName: Exit[174] Successors: []
NodeName: Entry[204] Successors: [Exit[205]]
NodeName: Exit[205] Successors: []
NodeName: Entry[196] Successors: [Exit[197]]
NodeName: Exit[197] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[35] Predecessor: [UncheckedExceptions[111], UncheckedExceptions[74], UncheckedExceptions[45], VariableDeclarator[107]]
NodeName: UncheckedExceptions[111] Predecessor: [UncheckedExceptions[74], VariableDeclarator[70]]
NodeName: UncheckedExceptions[74] Predecessor: [VariableDeclarator[54]]
NodeName: VariableDeclarator[54] Predecessor: [MethodAccess[44]]
NodeName: MethodAccess[44] Predecessor: [Entry[28]]
NodeName: Entry[28] Predecessor: []
NodeName: VariableDeclarator[70] Predecessor: [ParameterDeclaration[59]]
NodeName: ParameterDeclaration[59] Predecessor: [UncheckedExceptions[45]]
NodeName: UncheckedExceptions[45] Predecessor: [MethodAccess[44]]
NodeName: VariableDeclarator[107] Predecessor: [CatchParameterDeclaration[94]]
NodeName: CatchParameterDeclaration[94] Predecessor: [UncheckedExceptions[45]]
NodeName: Exit[123] Predecessor: [Entry[114]]
NodeName: Entry[114] Predecessor: []
NodeName: Exit[136] Predecessor: [Entry[135]]
NodeName: Entry[135] Predecessor: []
NodeName: Exit[128] Predecessor: [Entry[127]]
NodeName: Entry[127] Predecessor: []
NodeName: Exit[159] Predecessor: [Entry[158]]
NodeName: Entry[158] Predecessor: []
NodeName: Exit[151] Predecessor: [Entry[150]]
NodeName: Entry[150] Predecessor: []
NodeName: Exit[182] Predecessor: [Entry[181]]
NodeName: Entry[181] Predecessor: []
NodeName: Exit[174] Predecessor: [Entry[173]]
NodeName: Entry[173] Predecessor: []
NodeName: Exit[205] Predecessor: [Entry[204]]
NodeName: Entry[204] Predecessor: []
NodeName: Exit[197] Predecessor: [Entry[196]]
NodeName: Entry[196] Predecessor: []
